wsproto/frame_protocol.py

```diff
@@ -19,14 +18,21 @@
 try:
     from wsaccel.xormask import XorMaskerSimple  # type: ignore
 except ImportError:
+    _XOR_TABLE = [bytes(a ^ b for a in range(256)) for b in range(256)]
 
     class XorMaskerSimple:  # type: ignore
         def __init__(self, masking_key: bytes) -> None:
-            self._maskbytes = itertools.cycle(bytearray(masking_key))
+            self._masking_key = masking_key
 
         def process(self, data: bytes) -> bytes:
-            maskbytes = self._maskbytes
-            return bytearray(b ^ next(maskbytes) for b in bytearray(data))
+            if data:
+                data_array = bytearray(data)
+                a, b, c, d = (_XOR_TABLE[n] for n in self._masking_key)
+                data_array[::4] = data_array[::4].translate(a)
+                data_array[1::4] = data_array[1::4].translate(b)
+                data_array[2::4] = data_array[2::4].translate(c)
+                data_array[3::4] = data_array[3::4].translate(d)
+            return bytes(data_array)
```

Comment:
if `data` is empty, then `data_array` wouldn't exist at this point.

Reply:
Thanks